src/pages/teaching-plan-index/teaching-plan-index.ts

```diff
@@ -36,27 +34,39 @@ export class TeachingPlanIndexPage {
       teachingPlans.unities.forEach(unity => {
         let teachingPlans = [];
         unity.plans.forEach(plan => {
-          teachingPlans.push({ id: plan.id,
-                               description: plan.description + ' - ' + plan.grade_name });
+          teachingPlans.push({
+            id: plan.id,
+            description: plan.description + ' - ' + plan.grade_name
+          });
         });
-        this.unities.push({ name: unity.unity_name, teachingPlans: teachingPlans});
+        this.unities.push({ name: unity.unity_name, teachingPlans: teachingPlans });
       });
     });
   }
 
   toggleGroup(group) {
     if (this.isGroupShown(group)) {
-        this.shownGroup = null;
+      this.shownGroup = null;
     } else {
-        this.shownGroup = group;
+      this.shownGroup = group;
     }
   };
 
   isGroupShown(group) {
-      return this.shownGroup === group;
+    return this.shownGroup === group;
   };
 
   openDetail(teachingPlanId) {
     this.navCtrl.push(TeachingPlanDetailsPage, { "teachingPlanId": teachingPlanId });
   }
+
+  hasTeachingPlans(): boolean {
+    let hasPlans = false;
+    this.unities.forEach(unity => {
+      if (unity.teachingPlans && unity.teachingPlans.length > 0) {
+        hasPlans = true;
+      }
+    });
+    return hasPlans;
+  }
```

Comment:
Essa lógica não vai causar problema? 🤔 

O que aconteceria se a primeira unidade tiver `teachingPlans` e a última não?

Se for para não mostrar as unidades que não tem `teachingPlans`, eu colocaria a condicional dentro do `ngFor`.